packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -105,10 +104,16 @@ export class CoffeeCompiler extends CachingCompiler {
       output.js = doubleRoastedCoffee.data;
 
       if (doubleRoastedCoffee.sourceMap) {
+        // Reference the compiled CoffeeScript file so that `applySourceMap`
+        // below can match it with the source map produced by the CoffeeScript
+        // compiler.
+        doubleRoastedCoffee.sourceMap.sources[0] += '.js';
```

Comment:
Maybe set `doubleRoastedCoffee.sourceMap.sources[0] = this._outputFilePath(inputFile)`, if it's the same path (I'm pretty sure it is, but haven't checked)?